### PR TITLE
fix: ERR_UNSUPPORTED_ESM_URL_SCHEME on Windows (#215)

### PR DESCRIPTION
**Pull request by piccinnigius**
### Summary
see #215

### Test plan

- check if all runs good on Linux, I tested only on Windows, but I've added Os check to prevent different behaviors from before
- check if failing tests are related to two edits in start.ts and bundle.ts (sorry for this, I'm not so much into tests)

### CODE REVIEW

**Comment by piccinnigius** [comment]
i've also seen that linting has not worked :/

File: packages/repack/src/commands/bundle.ts
Comment (wodin) [inline packages/repack/src/commands/bundle.ts:8]: It looks like the reset of the project uses single quotes, so it seems like a bad idea to change them here.

File: packages/TesterApp/webpack.config.mjs
Comment (wodin) [inline packages/TesterApp/webpack.config.mjs:4]: The other files seem to have a blank line after the imports

File: packages/repack/src/commands/bundle.ts
Comment (wodin) [inline packages/repack/src/commands/bundle.ts:26]: Switching to tabs make it hard to see your actual changes. Also, since the rest of the project seems to be using two spaces for indents, I think switching to tabs is perhaps not a good idea

File: packages/repack/src/commands/start.ts
Comment (wodin) [inline packages/repack/src/commands/start.ts:18]: Quotes inconsistent with the existing code

**Comment by piccinnigius** [comment]
@wodin thanks for the feedback, how can i apply all this changes without change settings in my ide? i use intellij

sorry but i am a newbe in pull requests

**Comment by wodin** [comment]
@piccinnigius hmmm... sorry, I don't know IntelliJ.

Also, I am not in any way involved in this project. I am just watching it. But in general with PRs it's best to avoid changing spacing/quoting etc. at the same time as making other changes, since it obscures the real changes.

I see the project does not have a `.prettierrc` (or equivalent), but there is a [`.editorconfig`](https://github.com/callstack/repack/blob/d81c4992dc902eae0e89849a882e15056c08a61b/.editorconfig#L3-L10) which defines some things (but not the quoting style). Maybe IntelliJ can be made to adhere to `.editorconfig`?

**Comment by piccinnigius** [comment]
@wodin thanks a lot for infos. I'll find out. You know how to resubmit the same pull request with changes, or is better to close this and make a new one?

**Comment by wodin** [comment]
@piccinnigius I'm not sure what the norm is in this project, but no need to close this PR and open a new one. You can either just push more commits to the same branch to fix the issues, or you can rebase the branch with the changes to make it look like the original commits never existed, and then force push the branch.

If I look at a random merged PR it looks like they just fixed the linting issues in a new commit and didn't bother rebasing.
https://github.com/callstack/repack/pull/256/commits
But I didn't check other PRs to see if this is the norm. If you just add new commits to fix the issues, you can always rebase later before the PR is merged if the maintainers want you to do that.

By the way, I see there's [`lint` script in `package.json`](https://github.com/callstack/repack/blob/d81c4992dc902eae0e89849a882e15056c08a61b/package.json#L17), so you should at least be able to check the linting locally by running `yarn lint`.

**Comment by piccinnigius** [comment]
fixed linting problems

**Comment by edisonpaul4** [comment]
@wodin Any news regarding this PR?

We want to start using repack but is a deal breaker that repack doesnt work correctly on windows.

File: packages/repack/src/commands/bundle.ts
Comment (thymikee) [inline packages/repack/src/commands/bundle.ts:34]: rewriting `args.webpackConfig` is quite unexpected. Why do we need to do that? You can pass an absolute path to the CLI argument yourself.

Also `getWebpackConfigPath` should be responsible for providing us correct `webpackConfigPath` for any platform, so we don't need to override it (and repeat the code in `start` command). Appending the `file:\\` doesn't sound like an elegant solution here. Can you think about better way to achieve the correct path for Windows?

Comment (piccinnigius) [reply packages/repack/src/commands/bundle.ts:34]: I know that this is not the best solution. I'll try to figure out, any hint?

Comment (thymikee) [reply packages/repack/src/commands/bundle.ts:34]: @RafikiTiki I'd appreciate your help here

**Comment by piccinnigius** [comment]
hi @RafikiTiki, i've managed to reduce edits in multiple parts of repack to make it works by editing only getWebpackConfigPath and the dirname in tester app config. The last bad thing still the  need of `file://` prepending. do you have any idea how to avoid this platform specific issue? thanks in advance

**Review by RafikiTiki — APPROVED** [review]
Big thanks for your contribution @piccinnigius, and sorry it took me so long to finally review.

I recently got my hands on an old Windows machine, and (with some issues, but not related to the scope of this PR) I was able to run your branch successfully. I've also double-checked if this didn't have an impact on Mac users, and unsurprisingly – it didn't!

I'm merging it now, we plan to create a new release with @jbroma soon 🚀 

Edit: this probably can fix #440 #432 #415 #340 and #215

**Comment by piccinnigius** [comment]
> Big thanks for your contribution @piccinnigius, and sorry it took me so long to finally review.
> 
> I recently got my hands on an old Windows machine, and (with some issues, but not related to the scope of this PR) I was able to run your branch successfully. I've also double-checked if this didn't have an impact on Mac users, and unsurprisingly – it didn't!
> 
> I'm merging it now, we plan to create a new release with @jbroma soon 🚀

Thanks so much, i'm glad it works and to be part of this very usefull project <3

**Pull request merged**
